### PR TITLE
Add contributing doc to repo

### LINKED ISSUE #46: Need to add contributing doc

### PR DESCRIPTION
**Pull request by TrimBiggs**
Fixes #46

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+* [Technical Mailing List](http://lists.projectcalico.org/mailman/listinfo/calico-tech_lists.projectcalico.org)
+* [Slack Calico Users Channel](https://calicousers.slack.com) ([Sign up](https://calicousers-slackin.herokuapp.com))
+* IRC - [#calico](https://kiwiirc.com/client/irc.freenode.net/#calico)
+
+## Reporting issues
+
+Before raising an issue with the *Calico Libnetwork plugin*), please check for 
+duplicate issues, and read our [Troubleshooting](https://github.com/projectcalico/calico-containers/blob/master/docs/Troubleshooting.md)
+and our [Frequently Asked Questions](https://github.com/projectcalico/calico-containers/blob/master/docs/FAQ.md)
+documents.
```
Comment (robbrockbank) [inline CONTRIBUTING.md]: Maybe worth pointing at calico-containers calico with docker networking docs

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+If you have a question, please hop on to our IRC or Slack channels (see above).
+
+If you do need to raise an issue, please include any of the following that may
+be relevant to your problem (including too much information is never
+a bad thing):
+
+-  A detailed description of the problem.
```
Comment (robbrockbank) [inline CONTRIBUTING.md]: docker version is important here - and possibly get them to ps the docker daemon to check it was started with a cluster store.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+If you tests are failing the automated tests and you don't believe they should
+be, you may need to rebase your branch off the latest master.
+
+The unit tests can be run with:
+```
+make test
+```
+
+Where possible, please add any additional tests to ensure we maintain healthy
+code and feature coverage.
```
Comment (robbrockbank) [inline CONTRIBUTING.md]: I would also point at the main README for details on how to run the plugin for live testing.

**Comment by robbrockbank** [comment]
can you rebase?  couple of minor comments and then lgtm

File: CONTRIBUTING.md (hunk 2 above)
Comment (TrimBiggs) [inline CONTRIBUTING.md]: Done (below)

**Pull request merged**